pkg/controllers/vdb/upgrade.go

```diff
@@ -132,12 +132,26 @@ func (i *UpgradeManager) isVDBImageDifferent(ctx context.Context, sandbox string
 	if err != nil {
 		return false, err
 	}
+	clusterImage := ""
 	for inx := range stss.Items {
 		sts := stss.Items[inx]
 		cntImage, err := vk8s.GetServerImage(sts.Spec.Template.Spec.Containers)
 		if err != nil {
 			return false, err
 		}
+
+		// If the current cluster has different images across its subclusters, it means
+		// that some subclusters have just been unsandboxed. In this case, we do not want
+		// the upgrade reconciler to be triggered. Instead, the unsandbox-image-version reconciler
+		// should first restore the image version for those subclusters.
+		if clusterImage != "" && clusterImage != cntImage {
```

Comment:
Is it possible to test the case where you upgrade the main cluster in the middle of unsandboxing or just after?

Reply:
I remember you added a webhook to prevent online upgrade if there is an existing sandbox so we cannot upgrade the main cluster in the middle of unsandboxing. I added one test for upgrading after unsandboxing.